### PR TITLE
[RDY] Speed up some staff and rooms counts

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Copies the use of max count from hospital:countPatients into countStaffOfCategory and countRoomOfType
- Use the shortcut everywhere only a minimum staff/room requirement is called for

@Alberth289346 requested this in IRC recently.

### CODE REVIEW

File: CorsixTH/Lua/humanoid_actions/seek_reception.lua [hunk 1]
```diff
@@ -96,7 +96,7 @@ local function action_seek_reception_start(action, humanoid)
       end
     end
   local desks = #humanoid.hospital:findReceptionDesks()
-  local receptionists = humanoid.hospital:countStaffOfCategory("Receptionist")
+  local receptionists = humanoid.hospital:countStaffOfCategory("Receptionist", 1)
   if (receptionists > 1 and desks > 0) or (receptionists > 0 and desks > 1) then
```
Comment (Alberth289346) [inline CorsixTH/Lua/humanoid_actions/seek_reception.lua:100]: `receptionists > 1` won't work now anymore

File: CorsixTH/Lua/humanoid_actions/seek_reception.lua [hunk 2]
```diff
@@ -96,7 +96,7 @@ local function action_seek_reception_start(action, humanoid)
...
     local queueavg = math.floor(queuetotal / desks)
     if receptionists < desks and queueavg > 5 then
```
Comment (Alberth289346) [inline CorsixTH/Lua/humanoid_actions/seek_reception.lua:102]: `receptionists < desks` changes too, better just compute the actual number of receptionists I guess

**Comment by tobylane** [comment]
Corrected, thanks.

**Comment by Alberth289346** [comment]
Looks fine to me

**Review by lewri — APPROVED** [review]

**Review by TheCycoONE — APPROVED** [review]
LGTM, thanks. Do you notice the difference?

**Comment by tobylane** [comment]
No, but I don't have any mega hospitals now.

Ready.

**Pull request merged**